app/controllers/doorkeeper/authorizations_with_role_check_controller.rb

```diff
@@ -14,8 +14,8 @@ def new
 
     private
 
-    def resource_owner_has_role_for_application?
-      current_resource_owner.memberships.with_any_role(*application.available_role_names).exists?
+    def resource_owner_has_access_to_application?
+      current_resource_owner.memberships.with_access_to_application_name(application.name).exists?
```

Comment:
Slightly #pedantic - but would it make more sense for this to be called `with_access_to_application()`?